components/bl-weather-component/src/index.js

```diff
@@ -0,0 +1,62 @@
+import { useEffect, useState } from 'react';
+
+import ReactWeather, { useOpenWeather } from './lib/weather';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const UNITS = {
+  standard: { temperature: 'K', windSpeed: 'm/s' },
+  metric: { temperature: '°C', windSpeed: 'm/s' },
+  imperial: { temperature: '°F', windSpeed: 'mph' }
+};
+
+export default function WeatherComponent({ component, elRef, settings }) {
+  const { classList, style, display, currentLocation, location, label, forecast, unit, theme, lang, } = component;
+  const { API_KEY } = settings;
+
+  const [options, setOptions] = useState({ key: API_KEY, lat: null, lon: null, lang, unit });
+
+  useEffect(async () => {
+    let usedLocation;
+
+    if (currentLocation) {
+      usedLocation = await BackendlessUI.Navigator.getCurrentGeolocation();
+    } else if (location?.lat && location.lng) {
+      usedLocation = location;
+    } else {
+      console.error('Location data not provided or provided incorrectly');
+    }
```

Comment:
```suggestion
const usedLocation = currentLocation
   ? await BackendlessUI.Navigator.getCurrentGeolocation()
   : location
   
   if(usedLocation?.lat) {
     setOptions({ ...options, lat: usedLocation.lat, lon: usedLocation.lng });
   } else {
    console.error('Location data not provided or provided incorrectly');
   }
```

As I see, in prev code version we use setter only if location exists.